### PR TITLE
check if containerd is running not if system is running

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes: #2262

### CODE REVIEW

File: pkg/cluster/internal/create/actions/config/config.go [hunk 1]
```diff
@@ -185,8 +185,8 @@ func (a *Action) Execute(ctx *actions.ActionContext) error {
 					return errors.Wrap(err, "failed to write patched containerd config")
 				}
 				// restart containerd now that we've re-configured it
-				// skip if the systemd (also the containerd) is not running
-				if err := node.Command("bash", "-c", `! systemctl is-system-running || systemctl restart containerd`).Run(); err != nil {
+				// skip if containerd is not running
+				if err := node.Command("bash", "-c", `! pgrep --exact containerd || systemctl restart containerd`).Run(); err != nil {
```
Comment (BenTheElder) [inline pkg/cluster/internal/create/actions/config/config.go:189]: `pgrep` returns `0` on a match and `1` on no match.

```
EXIT STATUS
0      One or more processes matched the criteria. For pkill and
              pidwait, one or more processes must also have been
              successfully signalled or waited for.
1      No processes matched or none of them could be signalled.
2      Syntax error in the command line.
3      Fatal error: out of memory etc.
```
https://man7.org/linux/man-pages/man1/pgrep.1.html

**Comment by tao12345666333** [comment]
LGTM！

It seems more appropriate to check the process directly

fixes #2352

File: pkg/cluster/internal/create/actions/config/config.go (hunk 1 above)
Comment (BenTheElder) [reply pkg/cluster/internal/create/actions/config/config.go:189]: containerd when running has a process tree rooted by `containerd`, the child processes are the shim which is `containerd-shim`. `pgrep --exact [pattern]` matches only an exact pattern match.

Comment (aojea) [reply pkg/cluster/internal/create/actions/config/config.go:189]: Why not pkill -HUP then?

**Review by cpanato — COMMENTED** [review]
Nice

LGTM

File: pkg/cluster/internal/create/actions/config/config.go (hunk 1 above)
Comment (BenTheElder) [reply pkg/cluster/internal/create/actions/config/config.go:189]: This shell command we crafted exits success when successful in any case and only non zero when something doesn't work. We can correctly distinguish success.

Comment (aojea) [reply pkg/cluster/internal/create/actions/config/config.go:189]:
```suggestion
				if err := node.Command("bash", "-c", `pgrep --exact containerd || systemctl restart containerd`).Run(); err != nil {
```

Comment (aojea) [reply pkg/cluster/internal/create/actions/config/config.go:189]: we want to restart containerd if the pgrep fails, right? we don't need to negate then

Comment (aojea) [reply pkg/cluster/internal/create/actions/config/config.go:189]:
```
$ bash -c '! pgrep --exact containerd || echo "notfound"'
490904
notfound
$ bash -c 'pgrep --exact containerd || echo "notfound"
490904
$ bash -c 'pgrep --exact container || echo "notfound"'
notfound
```

Comment (BenTheElder) [reply pkg/cluster/internal/create/actions/config/config.go:189]: If the pgrep fails there is no containerd process found, so what are we restarting? 🙃

Comment (aojea) [reply pkg/cluster/internal/create/actions/config/config.go:189]: lol, what I was thinking ...

**Comment by aojea** [comment]
/lgtm
/approve
doooooh

**Pull request merged**